sycl/include/sycl/ext/oneapi/experimental/annotated_ptr/annotated_ptr.hpp

```diff
@@ -67,32 +67,27 @@ namespace {
 #define PROPAGATE_OP(op)                                                       \
   T operator op##=(const T &rhs) const { return *this = *this op rhs; }
 
-// compare strings on compile time
-constexpr bool compareStrs(const char *Str1, const char *Str2) {
-  return std::string_view(Str1) == Str2;
-}
-
-// filter properties with AllowedPropsTuple via name checking
-template <typename TestProps, typename AllowedPropsTuple>
-struct PropertiesAreAllowed {};
-
-template <typename TestProps, typename... AllowedProps>
-struct PropertiesAreAllowed<TestProps, std::tuple<const AllowedProps...>> {
-  static constexpr const bool allowed =
-      (compareStrs(detail::PropertyMetaInfo<TestProps>::name,
-                   detail::PropertyMetaInfo<AllowedProps>::name) ||
-       ...);
-};
-
 template <typename... Ts>
 using tuple_cat_t = decltype(std::tuple_cat(std::declval<Ts>()...));
 
+// filter properties with AllowedPropsTuple via key checking
 template <typename AllowedPropTuple, typename... Props>
 struct PropertiesFilter {
   using tuple = tuple_cat_t<typename std::conditional<
-      PropertiesAreAllowed<Props, AllowedPropTuple>::allowed, std::tuple<Props>,
+      detail::ContainsProperty<typename Props::key_t, AllowedPropTuple>::value,
+      std::tuple<Props>, std::tuple<>>::type...>;
+};
+
+// filter properties with Inheritable property
```

Comment:
I think "Inheritable" isn't great naming. 
Why is a single property sufficient? Don't we need one per property?